internal/log/log_test.go

```diff
@@ -140,6 +140,51 @@ func TestRecordLoggerIgnore(t *testing.T) {
 	assert.Contains(t, tp.Logs()[0], "appsec")
 }
 
+func TestSetLoggingRate(t *testing.T) {
+	defer func(old Logger) { UseLogger(old) }(logger)
+	tp := new(RecordLogger)
+	UseLogger(tp)
+
+	testCases := []struct {
+		value       string
+		expectedLog string
+	}{
+		{
+			value:       "",
+			expectedLog: "",
+		},
+		{
+			value:       "0",
+			expectedLog: "",
+		},
+		{
+			value:       "10",
+			expectedLog: "",
+		},
+		{
+			value:       "-1",
+			expectedLog: "Invalid value for DD_LOGGING_RATE: negative value",
```

Comment:
In general, I don't love tests that check exact log messages, since it can be frustrating to have to change the tests if we change the errors being logged even slightly. Plus it doesn't *really* test the outcome of the function call. Any way to just check the value of `errrate` instead?

Reply:
@katiehockman You are right, I'm changing this in a minute.